app/code/Magento/InventoryApi/Api/SourceItemRepositoryInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api;
+
+interface SourceItemRepositoryInterface
+{
+    /**
+     * Save Source item data. If you want to create plugin on get method, also you need to create separate plugin
+     * on getList method, because entity loading way is different for these methods
+     *
+     * @param \Magento\InventoryApi\Api\Data\SourceItemInterface $sourceItem
+     * @return int
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
+     */
+    public function save(\Magento\InventoryApi\Api\Data\SourceItemInterface $sourceItem);
```

Comment:
@maghamed  but than we need some bulk save function the following code also the same e  performance bottleneck.

`  /**
     * @inheritdoc
     */
    public function save(array $sourceItemList)
    {
        try {
            
             $savedSourceItems = [];
            foreach ($sourceItemList as $sourceItem)
            {
                $this->resourceSource->save($sourceItem);
                $savedItems[] = $sourceItem->getSourceItemId();
            }
            
            return $savedSourceItems; 
        } catch (\Exception $e) {
            $this->logger->error($e->getMessage());
            throw new CouldNotSaveException(__('Could not save source item'), $e);
        }
    }`